### PR TITLE
[fix] added ability to use * on Add (correct API) [add] Expose ParseRanges and ParseDataPoints

### PR DESCRIPTION
**Pull request by filipecosta90**
- [fix] added ability to use * on Add (correct API)
- [add] Expose ParseRanges and ParseDataPoints ( ParseInfo and ParseRules are already exposed )

### CODE REVIEW

**Comment by gkorland** [comment]
Let's go with two functions and not using interface

File: client.go [hunk 1]
```diff
@@ -292,7 +292,7 @@ type DataPoint struct {
 	Value     float64
 }
 
-func parseDataPoints(info interface{}) (dataPoints []DataPoint, err error) {
+func ParseDataPoints(info interface{}) (dataPoints []DataPoint, err error) {
```
Comment (danni-m) [inline client.go]: @filipecosta90 I don't object of exposing this, but I missing the reasoning of doing that.

Comment (filipecosta90) [reply client.go]: @danni-m I think the main reason is to decouple the client logic with response parsing logic. If the user wants to use their client and parse the response with our implementation they can ( we need it for TS.RANGE parsing ).

**Comment by danni-m** [comment]
@filipecosta90 I think we should avoid using `inteface{}` where we can.

**Comment by filipecosta90** [comment]
> @filipecosta90 I think we should avoid using `inteface{}` where we can.

agree with both you and @gkorland . Closing this PR and issuing a new one specifically so that we can discuss the new API

**Pull request closed without merging**